test/e2e_node/docker_test.go

```diff
@@ -70,4 +78,170 @@ var _ = framework.KubeDescribe("Docker features [Feature:Docker]", func() {
 			}
 		})
 	})
+
+	Context("when live-restore is enabled [Serial] [Slow] [Disruptive]", func() {
+		It("containers should not be disrupted when the daemon is down", func() {
+			const (
+				podName           = "live-restore-test-pod"
+				containerName     = "live-restore-test-container"
+				volumeName        = "live-restore-test-volume"
+				timestampFilename = "timestamp"
+			)
+
+			isSupported, err := isDockerLiveRestoreSupported()
+			framework.ExpectNoError(err)
+			if !isSupported {
+				framework.Skipf("Docker live-restore is not supported.")
+			}
+
+			By("Check whether live-restore is enabled.")
+			isEnabled, err := isDockerLiveRestoreEnabled()
+			framework.ExpectNoError(err)
+			if !isEnabled {
+				framework.Skipf("Docker live-restore is not enabled.")
+			}
+
+			// Creates a temporary directory that will be mounted into the
+			// container, serving as the communication channel between the host
+			// and the container.
+			By("Create temporary directory for mount.")
+			tempDir, err := ioutil.TempDir("", "")
+			framework.ExpectNoError(err)
+			defer func() {
+				By("Remove temporary directory.")
+				os.RemoveAll(tempDir)
+			}()
+
+			// Creates a container that writes the current timestamp every
+			// second to the timestamp file. We will be able to tell whether
+			// the container is running by checking if the timestamp increases.
+			cmd := "" +
+				"while true; do " +
+				"    date +%s > /test-dir/" + timestampFilename + "; " +
+				"    sleep 1; " +
+				"done"
+			By("Create the test pod.")
+			f.PodClient().CreateSync(&v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: podName},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{{
+						Name:    containerName,
+						Image:   "gcr.io/google_containers/busybox:1.24",
+						Command: []string{"/bin/sh"},
+						Args:    []string{"-c", cmd},
+						VolumeMounts: []v1.VolumeMount{
+							{
+								Name:      volumeName,
+								MountPath: "/test-dir",
+							},
+						},
+					}},
+					Volumes: []v1.Volume{{
+						Name: volumeName,
+						VolumeSource: v1.VolumeSource{
+							HostPath: &v1.HostPathVolumeSource{Path: tempDir},
+						},
+					}},
+				},
+			})
+
+			startTime1, err := getContainerStartTime(f, podName, containerName)
+			framework.ExpectNoError(err)
+
+			By("Stop Docker daemon.")
+			framework.ExpectNoError(stopDockerDaemon())
+			defer func() {
+				By("Restart Docker daemon.")
+				framework.ExpectNoError(startDockerDaemon())
+			}()
+
+			By("Ensure that the test container is running when Docker daemon is down.")
+			isRunning, err := isContainerRunning(filepath.Join(tempDir, timestampFilename))
+			framework.ExpectNoError(err)
+			if !isRunning {
+				framework.Failf("The container should be running but it's not.")
+			}
+
+			By("Ensure that the test pod is still running when Docker daemon is down.")
+			framework.ExpectNoError(f.WaitForPodRunning(podName))
```

Comment:
I agree that the behavior is not well defined so I have this test removed. But for now, the pod should be in running state, otherwise the live-restore feature would be less useful.